src/Libraries/Microsoft.Extensions.AI/Utilities/AIJsonUtilities.Schema.cs

```diff
@@ -278,24 +276,24 @@ JsonNode TransformSchemaNode(JsonSchemaExporterContext ctx, JsonNode schema)
                 {
                     objSchema.Add(AdditionalPropertiesPropertyName, (JsonNode)false);
                 }
-            }
-
-            if (ctx.Path.IsEmpty)
-            {
-                // We are at the root-level schema node, update/append parameter-specific metadata
 
                 // Some consumers of the JSON schema, including Ollama as of v0.3.13, don't understand
                 // schemas with "type": [...], and only understand "type" being a single value.
                 // STJ represents .NET integer types as ["string", "integer"], which will then lead to an error.
-                if (TypeIsArrayContainingInteger(schema))
+                if (TypeIsArrayContainingInteger(objSchema))
                 {
                     // We don't want to emit any array for "type". In this case we know it contains "integer"
                     // so reduce the type to that alone, assuming it's the most specific type.
-                    // This makes schemas for Int32 (etc) work with Ollama
+                    // This makes schemas for Int32 (etc) work with Ollama.
```

Comment:
Is this specific to Ollama? It feels wrong to me changing how we create schemas to cater to one specific (buggy?) implementation. Or is this a more general and reasonable restriction?

Reply:
Agreed. This was added as ad-hoc fix to unblock Ollama applications specifically until they fix it on their side (an issue has been filed for this).